addons/website/models/res_config_settings.py

```diff
@@ -106,7 +106,9 @@ def inverse_has_google_analytics_dashboard(self):
     def _onchange_language_ids(self):
         # If current default language is removed from language_ids
         # update the website_default_lang_id
-        if self.language_ids and self.website_default_lang_id not in self.language_ids:
+        if not self.language_ids:
+            self.website_default_lang_id = False
+        elif self.website_default_lang_id not in self.language_ids:
```

Comment:
Still it allows to select (BTW works well on Save) language which is not present on Available language.

![Screenshot from 2019-07-02 02-07-49](https://user-images.githubusercontent.com/16624719/60465250-49342300-9c6e-11e9-9eda-61f3830d359f.png)

Maybe missing domain there (and Limitation on selection widget )